### ISSUE TITLE
Refactor tests to use test-assembler

### ISSUE THREAD

**Issue by fitzgen** [gimli-rs/gimli#122]
It is more readable and less error prone (especially when updating tests and changing the length of some sequence of bytes!) than the old style of writing things out by hand.

We can do this piecemeal, no need to port all tests over at once.

See also #121, which will be very helpful for this work.

**Comment by philipc** [comment]
Mostly done. I think the main thing left is some examples in unit.rs. I'm not sure of the value of using those examples as tests. Maybe they can be simplified to code that simply compiles, rather than runs.

**Comment by fitzgen** [comment]
Agreed. Now that we have thorough unit tests, the doc tests can focus on
being examples, rather than tests.
